### DISCUSSION TITLE
OpenModelica 1.22.0-64bit installed exclusively for current user -- How to install the OpenModelica on Windows properly for all users?

### DISCUSSION THREAD

**Discussion by ialkov** [OpenModelica/OpenModelica#11635] [Q&A]
For several releases of OpenModelica, I am facing an annoying behavior of the installer. Let it be the installer of most recent official release OpenModelica-v1.22.0-64bit.exe.

If I simpliy execute the installer from Windows Explorer and follow the guided installation steps, OpenModelica is installed exclusively for the user, whose admin permissions are applied and who is currently logged on Windows. I.e. all other users of the Windows system do not get any start entries for OpenModelica in the Windows programs menu. Furthermore, OpenModelica gives some error messages, I cannon interpret due to my knowledge gaps :-)
```
[9] 11:16:00 Error Curl error for URL https://libraries.openmodelica.org/index/v1/index.json: Timeout was reached
[10] 11:16:00 Scripting Error Failed to download package index https://libraries.openmodelica.org/index/v1/index.json to file C:/Users/[myuser]/AppData/Roaming/.openmodelica/libraries/index.json
```

If I execute the installer from cmd in silent mode (option "/S"), it is installed properly for all users, but I naturally cannot adapt the installation directory. Also, not by the option "/D my_path". But I mandatory need to adapt the installation directory pointing to on second drive of the machine "D:\Program_Files".

Best way would be, if OpenModelica implements in the installer the option for installation for all users available within the installer GUI. Until then, I hope you could help me with some hints please.

**Comment by casella** [comment]
@adeas31, @adrpo could we install OMC for all users?

@ialkov, the installer tries to reach our server to download the latest index file. Are you disconnected from the internet when you try to install?

**Reply by ialkov** [reply]
@casella, Yeas, I am permanently disconnected from internet on the system where I have to install OpenModelica. To be more precise, I would like to install the complete OpenModelica environments and tools as it is intended by the installer. Not just the OMC.

**Comment by casella** [comment]
OK, in that case you should anyway get the Modelica Standard Library installed from a cache, for all the OpenModelica environment.

**Reply by ialkov** [reply]
I think you are right. I have the MSL installed by the installer itself on this system.

**Reply by casella** [reply]
Sorry about the error, maybe we should catch it and replace it with something nicer like "Cannot connect to the remote server to download libraries. Installing Modelica libraries from local cache".

**Reply by ialkov** [reply]
Those two warning I mention above
```
[9] 11:16:00 Error Curl error for URL https://libraries.openmodelica.org/index/v1/index.json: Timeout was reached
[10] 11:16:00 Scripting Error Failed to download package index https://libraries.openmodelica.org/index/v1/index.json to file C:/Users/[myuser]/AppData/Roaming/.openmodelica/libraries/index.json
```
are only present if I install the OpenModelica in GUI mode of installer. 


But if I install the OpenModelica from cmd in silent mode, there seems to be all right (cf. just two messages below, nothing more), besides the standard installation path I am not allowed to use.
```
[1] 16:38:33 Scripting Notification
Automatically loaded package Complex 4.0.0 due to uses annotation from Modelica.

[2] 16:38:33 Scripting Notification
Automatically loaded package ModelicaServices 4.0.0 due to uses annotation from Modelica.
```
